pkg/boot/kexec/kexec_load_linux.go

```diff
@@ -65,3 +95,182 @@ func rawLoad(entry uintptr, segments []Segment, flags uint64) error {
 	}
 	return nil
 }
+
+// KexecLoad loads the given kernel file as to-be-kexeced kernel with
+// the given ramfs file and cmdline string. It uses the kexec "classic"
+// system call, i.e. memory segments + entry point.
+// Arguably, the args should be io.ReaderAt and io.Reader, but we match
+//the existing Load function. If you *do* decided to make these io.Reader,
+// beware of the code that formerly passed a *os.File, and then checked
+// to see if the ramfs (io.Reader) was != nil
+// that won't work, see when-nil-isnt-equal-to-nil for why.
+// Comparison to nil in general is not a great practice, but one fix
+// at a time.
+func KexecLoad(kernel, ramfs *os.File, cmdline string) error {
+	bzimage.Debug = Debug
+
+	/* A collection of vars used for processing the kernel for kexec */
+
+	var err error
+	// b is the deserialized bzImage from the kernel
+	// io.ReaderAt.
+	var b bzimage.BzImage
+	// kmem is a struct holding kexec segments.
+	//
+	// It has routines to work with physical memory
+	// ranges.
+	var kmem *Memory
+
+	bp, err := ioutil.ReadFile("/sys/kernel/boot_params/data")
+	if err != nil {
+		return fmt.Errorf("Reading boot_param data: %w", err)
```

Comment:
do not capitalize error messages

fmt.Errorf("reading ...

here and everywhere

Reply:
Done.

Reply:
doesn't look done